oak_functions/wasm/src/lib.rs

```diff
@@ -136,90 +137,133 @@ where
         extensions: HashMap<ExtensionHandle, Box<dyn OakApiNativeExtension>>,
     ) -> anyhow::Result<Self> {
         let user_state = UserState::init(request_bytes, extensions, logger);
-
         // For isolated requests we need to create a new store for every request.
         let mut store = wasmi::Store::new(module.engine(), user_state);
+        let linker = OakLinker::new(module.engine(), &mut store);
+        let (instance, store) = linker.instantiate(store, module)?;
+        // TODO(#3785): Add logging and list exports.
+        let wasm_state = Self { instance, store };
+        Ok(wasm_state)
+    }
 
-        let mut linker: wasmi::Linker<UserState<L>> = wasmi::Linker::new(module.engine());
+    // Invokes the Wasm module by calling main.
```

Comment:
nit: pre-existing, but comments directly above a function, struct, etc should be doccomments, event if it is not public.  AFAIK we also generate documentation for non-public code.

Reply:
Great, I'll refactor some more in the next PRs and do this then.